app/model/IDatabaseWrapper.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Michal Struna
+ * Date: 24/11/2018
+ * Time: 16:16
+ */
+
+namespace App\Model;
+
+interface IDatabaseWrapper {
+
+    /**
+     * Select items from DB.
+     * @return array Array of DB items.
+     */
+    public function getAll(): array; // TODO: Array of selected columns in parameter?
```

Comment:
JJ to si myslim ze bychom meli udelat... a melo by to jit normalne tema 3 teckama (btw muzem zvysit tu min verzi php na 7.2) https://stackoverflow.com/questions/45419150/php-spread-operator-in-array-declaration

Reply:
Problém je v tom, že PDO nepodporuje prepared statements pro názvy sloupců (pouze pro jejich hodnoty). Takže skrz název sloupce se může udělat SQL injekce. Na druhou stranu, název sloupce nezadává uživatel, ale bude napevno v kódu, takže to asi vadit nebude.

Reply:
Aha 👀 ... ale jn to bude v kodu